autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/feature_preprocessing/ExtraTreesPreprocessorClassification.py

```diff
@@ -0,0 +1,157 @@
+
+
+class ExtraTreesPreprocessorClassification(autoPyTorchFeaturePreprocessingComponent):
+    """
+    Select features based on importance weights calculated using extra trees
+    """
+    def __init__(self, bootstrap: bool = True, n_estimators: int = 10,
+                 criterion: str = "gini", max_features: float = 0.5,
+                 max_depth: Optional[Union[str, int]] = 5, min_samples_split: int = 2,
+                 min_samples_leaf: int = 1, min_weight_fraction_leaf: float = 0,
+                 max_leaf_nodes: Optional[Union[str, int]] = "none",
+                 min_impurity_decrease: float = 0, oob_score: bool = False,
+                 verbose: int = 0,
+                 random_state: Optional[np.random.RandomState] = None):
+        self.bootstrap = bootstrap
+        self.n_estimators = n_estimators
+        if criterion not in ("gini", "entropy"):
+            raise ValueError("'criterion' is not in ('gini', 'entropy'): "
+                             "%s" % criterion)
+        self.criterion = criterion
+        self.max_features = max_features
+        self.min_impurity_decrease = min_impurity_decrease
+        self.max_depth = max_depth
+        self.min_samples_split = min_samples_split
+        self.min_samples_leaf = min_samples_leaf
+        self.min_weight_fraction_leaf = min_weight_fraction_leaf
+        self.max_leaf_nodes = max_leaf_nodes
+        self.oob_score = oob_score
+        self.verbose = verbose
+
+        super().__init__(random_state=random_state)
+
+    def fit(self, X: Dict[str, Any], y: Any = None) -> BaseEstimator:
+
+        if check_none(self.max_leaf_nodes):
+            self.max_leaf_nodes = None
+        elif isinstance(self.max_leaf_nodes, int):
+            self.max_leaf_nodes = int(self.max_leaf_nodes)
+        else:
+            raise ValueError(f"Expected `max_leaf_nodes` to be either "
+                             f"in ('None', 'none', None) or an integer, got {self.max_leaf_nodes}")
+
+        if check_none(self.max_depth):
+            self.max_depth = None
+        elif isinstance(self.max_depth, int):
+            self.max_depth = int(self.max_depth)
+        else:
+            raise ValueError(f"Expected `max_depth` to be either "
+                             f"in ('None', 'none', None) or an integer, got {self.max_depth}")
+
+        # TODO: add class_weights
+        estimator = ExtraTreesClassifier(
+            n_estimators=self.n_estimators,
+            criterion=self.criterion,
+            max_depth=self.max_depth,
+            min_samples_split=self.min_samples_split,
+            min_samples_leaf=self.min_samples_leaf,
+            bootstrap=self.bootstrap,
+            max_features=self.max_features,
+            max_leaf_nodes=self.max_leaf_nodes,
+            min_impurity_decrease=self.min_impurity_decrease,
+            oob_score=self.oob_score,
+            verbose=self.verbose,
+            random_state=self.random_state,
```

Comment:
Can we make a method that returns those attributes as a dict?
It will look like
```
    def component_kwargs(self) -> Dict[str, Any]:
        return dict(
            n_estimators=self.n_estimators,
            criterion=self.criterion,
            max_depth=self.max_depth,
            min_samples_split=self.min_samples_split,
            min_samples_leaf=self.min_samples_leaf,
            bootstrap=self.bootstrap,
            max_features=self.max_features,
            max_leaf_nodes=self.max_leaf_nodes,
            min_impurity_decrease=self.min_impurity_decrease,
            oob_score=self.oob_score,
            verbose=self.verbose,
            random_state=self.random_state,
    )
```

Reply:
How does it help in debugging?

Reply:
We can especially when we forget something, it will be easier to compare with attributes defined in `__init__` because we can make them closer in terms of location

Reply:
sure, that makes sense. I'll add them